src/watch/index.ts

```diff
@@ -197,25 +192,35 @@ export class Task {
 		}
 
 		return rollup(options)
-			.then((chunk: OutputChunk) => {
+			.then((result: OutputChunk | OutputChunkSet) => {
 				if (this.closed) return;
 
-				this.cache = chunk;
+				const watched = this.watched = new Set();
 
-				const watched = new Set();
-
-				chunk.modules.forEach((module: ModuleJSON) => {
-					watched.add(module.id);
-					this.watchFile(module.id);
-				});
+				const watchChunk = (chunk: {modules: ModuleJSON[]}) => {
+					chunk.modules.forEach((module: ModuleJSON) => {
+						watched.add(module.id);
+						this.watchFile(module.id);
+					});
 
-				this.watched.forEach(id => {
-					if (!watched.has(id)) deleteTask(id, this, this.chokidarOptionsHash);
-				});
+					this.watched.forEach(id => {
+						if (!watched.has(id)) deleteTask(id, this, this.chokidarOptionsHash);
+					});
+				}
 
-				this.watched = watched;
+				if ((<OutputChunkSet>result).chunks) {
+					const chunks = (<OutputChunkSet>result).chunks;
+					this.cache = { chunks };
+					for (const chunkName in chunks) {
+						watchChunk(chunks[chunkName]);
+					}
+				} else {
+					const chunk = (<OutputChunk>result);
+					this.cache = { chunks: { [this.inputOptions.entry]: chunk } };
```

Comment:
If the cache consists of `chunks` in both cases, do we still need the `this.cache.modules` check in https://github.com/rollup/rollup/pull/2035/files#diff-7c1e6304262c178e7fe268b62060e26bR85 ?

Reply:
Ah, good question.  I should have explained this in the code.

`Task#chunk` is typed as `{chunks: { [name]: { modules } } }` even in the single-bundle case just to keep [this code](https://github.com/rollup/rollup/pull/2035/files#diff-093a6241a0003efb710688c0b3332f53R237) slightly simpler.  

However, for the type of [`InputOptions.cache`](https://github.com/rollup/rollup/pull/2035/files#diff-c5fd8e65630bd641bf07130346ee8778R66), which is part of the public JS API, I went with `{ modules } | {chunks: { [name]: { modules } } }` so that this wouldn't be a breaking change (and because I think the existing `cache` option, wherein a developer can simply pass the previous bundle result back in as `cache:`, seems to make sense).

Reply:
> Task#cache is typed as {chunks: { [name]: { modules } } } even in the single-bundle case just to keep this code slightly simpler.

On second thought, I think it's probably clearer and easier to understand in the future if we just keep `Task#cache` typed identically to `InputOptions#cache`.  I'll push that change.